frontera/worker/strategy.py

```diff
@@ -54,7 +54,7 @@ def __init__(self, states):
         self._fingerprints = set()
 
     def to_fetch(self, requests):
-        if isinstance(requests, Sequence):
+        if isinstance(requests, Iterable):
             self._fingerprints.update([x.meta[b'fingerprint'] for x in requests])
```

Comment:
`set.update()` receives iterable too. You could save bit of memory here converting list to generator (i.e. remove `[` and `]`).